### PR TITLE
done

### PR DESCRIPTION
**Pull request by Hibaber**
Hola, 

he debido de cometer un o varios errores de ortografía y el navegador del test no me deja ver los errores 

Gracias

### CODE REVIEW

**Comment by Hibaber** [comment]
Hola,

he debido de cometer un o varios errores de ortografía y el navegador del test no me deja ver los errores

Gracias

**Review by ta-web-mad — COMMENTED** [review]
Hola Hiba!

No te ejecutaban los tests porque al inicio tenías una línea de código que no reconocía e impedía que se ejecutaran bien los tests:

`const arr = require("jshint/data/non-ascii-identifier-start");`

En la línea 137 tenías `count=+`, que JavaScript no reconoce. Si querías añadir 1 al valor, puedes hacer `count++` o lo que es lo mismo `count = count+1`

Sobre el resto de ejercicios, te dejo un par de comentarios para que tengas en cuenta 😉

File: src/functions-and-arrays.js [hunk 1]
```diff
@@ -1,21 +1,46 @@
...
-const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
+  if (words[i].length>largeWord.length) {
+    largestWord=strings[i];
+  }
+}
+
+}
+
```
Comment (ta-web-mad) [inline src/functions-and-arrays.js:31]: - Aquí te falta cubrir algunos corner cases que vienen en los test, como que por ejemplo devuelva null cuando se introduce un array vacío, lo que se solucionaría con un 
`if(words.length === 0) { return null }`
- Además, la función no retorna nada, te dejo como lo habría hecho yo para que puedas comparar (además intentas asignar `largestWord` a `strings[i]`, pero no tienes ninguna variable llamada `strings`):

```suggestion
function findLongestWord(words) {

  if (words.length === 0) { return null }

  let largestWord = ""

  for (let i = 0; i < words.length; i++) {
    if (words[i].length > largestWord.length) {
      largestWord = words[i];
    }
  }

  return largestWord;

}
```

File: src/functions-and-arrays.js [hunk 2]
```diff
@@ -1,21 +1,46 @@
...
+// Iteration #3: Calculate the sum
+const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
+
+function sumNumbers(numbers) {
+let sum=0;
+for (let i=0; i<numbers.length;i++){
+  sum+= numbers[i];
+}
+return sum;
+}
```
Comment (ta-web-mad) [inline src/functions-and-arrays.js:43]: - Aquí si funcionaría el `sum+= numbers[i]` porque le estás sumando un número a la derecha de la expresión 😉

File: src/functions-and-arrays.js [hunk 3]
```diff
@@ -26,13 +51,29 @@ function sum() {}
...
+function averageWordLength(wordsArr) {
+
+ let sum =0;
+  for (let i=0; i<arr.length;i++){
+    sum+=wordsArr[i].length;
+  }
+  return sum /  wordsArr.length;
+}
+
+
```
Comment (ta-web-mad) [inline src/functions-and-arrays.js:76]: - Aquí solo te ha faltado cubrir el caso de que si el array está vacío, la función retorne null y que en la línea 70 has llamado a `arr.length`! Pero en tu caso tu array se llama `wordsArr`, por lo tanto no lo reconoce:

```suggestion
function averageWordLength(wordsArr) {

 if (wordsArr.length ===  0) { return null }

 let sum =0;
  for (let i=0; i<wordsArr.length;i++){
    sum+=wordsArr[i].length;
  }
  return sum /  wordsArr.length;
}
```

File: src/functions-and-arrays.js [hunk 4]
```diff
@@ -52,16 +93,26 @@ const wordsUnique = [
...
 
-function uniquifyArray() {}
+function uniquifyArray(wordsUnique) {
+  let arr=[]
+for (let i=0; i<wordsUnique.length;i++){
+    if(wordsUnique.indexOf([i]!== -1)) {
+      return arr.push([i])
+    }
+  }
+}
```
Comment (ta-web-mad) [inline src/functions-and-arrays.js:103]: - Aquí no te funciona, porque cuando comparas el `indexOf()` con -1, indica que esa palabra está en el array, pero una palabra que está duplicada siempre te va a dar un valor diferente a -1, puesto que **sí** está en el array.
- La clave la tienes en que el `indexOf` siempre te devuelve la primera ocurrencia del elemento, por lo que si un elemento se repite varias veces, solo te devolvera la primera posición en la que ese elemento aparezca, que debería ser la misma que la de la `i` del bucle for que has definido, así que ahí tienes la clave ya que si los comparas y no son iguales, significará que ese elemento no es la primera vez que aparece!
